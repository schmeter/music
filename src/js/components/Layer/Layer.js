import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Icon from '../Icon';
import i18n from '../../services/i18n';

class Layer extends React.Component {
    render() {
        const { className, id, activeId, closeLayers, children } = this.props;

        return (
            <div
                className={classNames('layer', `layer-${className || id}`, {
                    closed: activeId !== id,
                })}
            >
                <header className="layer-header">
                    <h2 className="layer-headline">
                        {i18n(`layer_${id}_headline`)}
                    </h2>
                    <button
                        className="header-button close-button"
                        onClick={closeLayers}
                        tabIndex="-1"
                    >
                        <Icon id="close" />
                    </button>
                </header>
                {children}
            </div>
        );
    }
}

Layer.propTypes = {
    id: PropTypes.string.isRequired,
    className: PropTypes.string,
    children: PropTypes.node,
    activeId: PropTypes.string.isRequired,
    closeLayers: PropTypes.func.isRequired,
};

export default Layer;
