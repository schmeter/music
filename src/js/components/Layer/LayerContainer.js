import { connect } from 'react-redux';

import Layer from './Layer';
import { closeLayersAction } from '../../state/actions/layer';
import { getActiveId } from '../../state/selectors/layer';

export const mapDispatchToProps = dispatch => ({
    closeLayers: () => dispatch(closeLayersAction()),
});

export const mapStateToProps = state => ({
    activeId: getActiveId(state),
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Layer);
