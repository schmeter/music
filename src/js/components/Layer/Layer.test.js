import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Layer from './Layer';

const mockedEvent = {
    preventDefault: jest.fn(),
};

it('renders correctly', () => {
    const props = {
        id: 'test',
        className: 'test',
        children: 'test',
        activeId: 'test2',
        closeLayers: jest.fn(),
    };
    const component = shallow(<Layer {...props} />);

    component.find('button').simulate('click', mockedEvent);

    expect(toJson(component)).toMatchSnapshot();
    expect(props.closeLayers).toHaveBeenCalled();
});

it('renders correctly with id only', () => {
    const props = {
        id: 'test',
        activeId: 'test2',
        closeLayers: jest.fn(),
    };
    const component = shallow(<Layer {...props} />);

    expect(toJson(component)).toMatchSnapshot();
});
