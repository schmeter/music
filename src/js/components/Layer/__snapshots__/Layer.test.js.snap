// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="layer layer-test closed"
>
  <h2
    className="headline"
  >
    {{de:layer_test_headline}}
  </h2>
  <button
    className="header-button close-button"
    onClick={[MockFunction]}
  >
    <Icon
      id="close"
    />
  </button>
  test
</div>
`;
